packages/ui-components/src/components/DataTable/useDataTableExport.js

```diff
@@ -25,7 +25,12 @@ export const useDataTableExport = (columns, data, title, startDate, endDate) =>
     );
     const body =
       tableData.length > 0
-        ? tableData.map(row => tableColumns.map(col => row.values[col.id]))
+        ? tableData.map(row =>
+            tableColumns.map(col => {
+              const value = row.values[col.id];
+              return isNaN(value) ? value : Number(value);
```

Comment:
`isNaN()` has a surprising behaviour. The logic provided here will convert empty string '', false, null..etc to 0 because `isNaN` use `Number()` to validate the type and Number(false) = 0, so isNaN(false) = false.. 

Here I suggest we can do something like:
```
function convertToNumber(str) {
  // Try to parse the string as a floating-point number
  const num = parseFloat(str);

  // Check if the parsed value is a number (not NaN)
  if (!isNaN(num)) {
    return num;
  }

  // If the parsed value is NaN, return the original string
  return str;
}

Reply:
Good point, @biaoli0 , thanks for the suggestion